src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_app_extension.cc

```diff
@@ -72,24 +76,60 @@ bool VehicleInfoAppExtension::unsubscribeFromVehicleInfo(
 
 void VehicleInfoAppExtension::unsubscribeFromVehicleInfo() {
   LOG4CXX_AUTO_TRACE(logger_);
+  sync_primitives::AutoLock lock(*subscribed_data_lock_);
   subscribed_data_.clear();
 }
 
 bool VehicleInfoAppExtension::isSubscribedToVehicleInfo(
     const std::string& vehicle_data) const {
   LOG4CXX_DEBUG(logger_, vehicle_data);
+  sync_primitives::AutoLock lock(*subscribed_data_lock_);
   return subscribed_data_.find(vehicle_data) != subscribed_data_.end();
 }
 
-const VehicleInfoSubscriptions& VehicleInfoAppExtension::Subscriptions() {
-  return subscribed_data_;
+const DataAccessor<VehicleInfoSubscriptions>
+VehicleInfoAppExtension::Subscriptions() {
+  DataAccessor<VehicleInfoSubscriptions> data_accessor(subscribed_data_,
+                                                       subscribed_data_lock_);
+  return data_accessor;
+}
+
+bool VehicleInfoAppExtension::AddPendingSubscription(
+    const std::string& vehicle_data) {
+  sync_primitives::AutoLock lock(*pending_subscriptions_lock_);
+  return pending_subscriptions_.insert(vehicle_data).second;
+}
+
+bool VehicleInfoAppExtension::RemovePendingSubscription(
+    const std::string& vehicle_data) {
+  LOG4CXX_DEBUG(logger_, vehicle_data);
+  sync_primitives::AutoLock lock(*pending_subscriptions_lock_);
+  auto it = pending_subscriptions_.find(vehicle_data);
+  if (it != pending_subscriptions_.end()) {
+    pending_subscriptions_.erase(it);
+    return true;
+  }
+  return false;
+}
+
+bool VehicleInfoAppExtension::RemovePendingSubscriptions() {
```

Comment:
@LuxoftAKutsan does it make sense to return bool for this function?

Reply:
fixed